bpf/lib/process.h

```diff
@@ -262,6 +262,10 @@ struct msg_execve_event {
 		struct msg_process process;
 		char buffer[PADDED_BUFFER];
 	};
+	/* below fields are not part of the event, serve just as
+	 * heap for execve programs
+	 */
+	__u32 binary;
```

Comment:
Wouldn't be possible to create a new struct keep it local to bpf_execve_event.c file that has both msg_execve_event and the binary embedded inside?

Just to keep msg_execve_event clean and for its purpose and maybe it reduce errors or confusions, yeh you commented it ;-) but we may also start doing same in future and it can get ugly !

Reply:
I mark this resolved since we already do this for kprobes.